src/main.cpp

```diff
@@ -25,20 +31,27 @@ int main(int argc, char *argv[])
 
     QQmlApplicationEngine engine;
 
+QScopedPointer<OAuth2Config> authConfig(new OAuth2Config());
+qmlRegisterSingletonInstance("QtPygo.authConfig", 1, 0, "AuthConfig", authConfig.get());
 #ifdef OAUTH2
 //    TODO: remove hardcoded client id for OAuth2 authentication
-    QScopedPointer<Auth> auth(new Auth("devops-console"));
+    QScopedPointer<Auth> auth(new Auth(authConfig));
+    qmlRegisterSingletonInstance("QtPygo.auth", 1, 0, "Auth", auth.get());
+#endif
+
+#ifdef WASM_OAUTH2
+    QScopedPointer<WASMAuth> auth(new WASMAuth("devops-console"));
     qmlRegisterSingletonInstance("QtPygo.auth", 1, 0, "Auth", auth.get());
 #endif
 
     qmlRegisterType<AsyncSettings>("QtPygo.storage", 1, 0, "Settings");
 
 #ifdef Q_OS_WASM
```

Comment:
if and else code are identical.

As GenericEntry needs Auth, it should be set only in a if defined OAUTH2 or WASM_OAUTH2 section ?
if not defined the WebEntry can still be used.

WebEntry should be renamed GuestEntry or something like that and maybe GenericEntry should be renamed AuthEntry as this is the purpose of this PR.

In a next step (or now ?) we should remove GuestEntry and permit Guest connection from GenericEntry/AuthEntry if this is accepted by the backend ?